### PR TITLE
1877: Stockholm Tramways implemented auctioning of corporations and starting phases of corporations.

### PR DESCRIPTION
**Pull request by lbrande**
I had to fix a bug in `buy_sell_par_shares_via_bid.rb` to make it also work with games with a `HOME_TOKEN_TIMING` other than `:par`. This required changing the `HOME_TOKEN_TIMING` of 1861/1867 to `:par`, as it was incorrectly set to `:float` and it only worked correctly due to the bug I fixed.

### CODE REVIEW

**Review by perwestling — COMMENTED** [review]
What is s1?

File: lib/engine/game/g_1877_stockholm_tramways/game.rb [hunk 1]
```diff
@@ -139,10 +143,58 @@ class Game < Game::Base
            'sl_trigger' => ['SL Trigger', 'SL will form at end of OR, game ends at end of following OR set'],
          ).freeze
 
+        def setup
+          @sl = nil
```
Comment (perwestling) [inline lib/engine/game/g_1877_stockholm_tramways/game.rb:147]: What is s1?

Comment (lbrande) [reply lib/engine/game/g_1877_stockholm_tramways/game.rb:147]: SL is the name of the end of game event, like LNER in 1862. I have however not implemented it's logic yet.

**Comment by lbrande** [comment]
> What is s1?

See my answer above.

**Pull request merged**